### DISCUSSION TITLE
Doubt Regarding Data Augmentation in Yolov8

### DISCUSSION THREAD

**Discussion by SundarakrishnanN** [ultralytics/ultralytics#5950] [Q&A]
Hi,I didn't get better results on augmenting my data using the Yolo Arguments even though I used only the relevant options. So it got me thinking, does Yolo just rotate or flip the existing images and train on them or does it train on the augmented images along with the original ones?

**Comment by ExtReMLapin** [comment]
It trains alongs originals if the prob in yaml hyp config aren't at 1.0

**Reply by pderrenger** [reply]
@ExtReMLapin that's correct! YOLOv8 applies data augmentations such as rotation, flipping, scaling, etc., to the training images on-the-fly during each epoch. The augmented images are used in addition to the original images, not as replacements, as long as the probabilities for these augmentations (specified in the hyperparameter configuration file) are set to values less than 1.0. This approach helps the model generalize better by learning from a more diverse set of training examples. 😊👍
